Core/src/main/java/com/plotsquared/core/plot/PlotId.java

```diff
@@ -110,7 +110,7 @@ private PlotId(final int x, final int y) {
      * @return Plot ID
      */
     public static @NonNull PlotId unpair(final int hash) {
-        return PlotId.of(hash >> 16, hash & 0xFFFF);
+        return PlotId.of(hash >> 16, (short) (hash & 0xFFFF));
```

Comment:
Preferably deprecate the existing method and create a new one named `shortUnpair` or something that actually casts to a short.

Reply:
Fixing it like that is fine I guess, because it's a bug if it didn't work correctly, but I'd like to have more insights on under which circumstances this was discovered (and especially why it wasn't an issue before)

Reply:
Found in the inbox list
![grafik](https://user-images.githubusercontent.com/27054324/146256592-d97eb31c-5edc-4d79-8ccf-c989c97c8888.png)

Debugged the bit shifting etc and found out, that signed / negative coordinates are not correctly unpaired, because they exceed the bound for signed shorts. Therefor it is a bug. The method is only used in inboxes, maybe they are not used that often

Reply:
> Preferably deprecate the existing method and create a new one named shortUnpair or something that actually casts to a short.

Based on @SirYwell's comment and the intention of the method I would not create a new method for that. The current implementation is simply broken and shouldn't be extended with another one for the sake of fixing a fundamentally bug. 
The method is only used internally for the inbox. If other plugins are using this method, they will not be broken, as the hash generation is correct, only the unpairing for displaying the hash returns wrong data.